<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • PopGenHelpR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PopGenHelpR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../reference/index.html">Manual</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/PopGenHelpR_vignette.html">Vignette</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_heterozygosity.html">Choosing heterozygosity</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_PCA.html">Principal Component Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_createQmatrix.html">Creating a Q-matrix</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_sortQmatrix.html">Sorting a Q-matrix</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_MakingMaps.html">Making Maps</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_plotformatting.html">Formatting plots</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_benchmarking.html">Testing PopGen statistics</a></li>
    <li><a class="dropdown-item" href="../articles/PGH_viztests.html">Testing plotting arguments</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/kfarleigh/PopGenHelpR"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Changelog</h1>
      <small>Source: <a href="https://github.com/kfarleigh/PopGenHelpR/blob/main/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.4.0" id="popgenhelpr-140">PopGenHelpR 1.4.0<a class="anchor" aria-label="anchor" href="#popgenhelpr-140"></a></h2>
<ul><li>July 2nd, 2025 ## <code>Differentiation</code> and <code>Heterozygosity</code>
</li>
<li>Functions can now detect whether “|” or “" separate genotypes. This was causing functions to throw NAs.</li>
<li>Minor bug was fixed for Jost’s D calculation. A parentheses was removed in the calculation of the <code>D.bar</code> object.</li>
</ul><div class="section level3">
<h3 id="ancestry_barchart-1-4-0"><code>Ancestry_barchart</code><a class="anchor" aria-label="anchor" href="#ancestry_barchart-1-4-0"></a></h3>
<ul><li>Code has been simplified, individual and population information is now coerced to character before visualization.</li>
<li>It is now an error if the individual names in the population data do not match the names in the individual data.</li>
<li>Plots are automatically sorted by cluster so that individuals and populations are plotted with others in the cluster, if there is no specified order.</li>
<li>Users can specify the individual and population order using the <code>ind.order</code> and <code>pop.order</code> arguments, respectively.</li>
<li>Users can specify the legend position using the <code>legend_pos</code> argument.</li>
</ul></div>
<div class="section level3">
<h3 id="mapping-functions-1-4-0">Mapping functions<a class="anchor" aria-label="anchor" href="#mapping-functions-1-4-0"></a></h3>
<p>The following updates laregly apply for the <code>Plot_coordinates</code>, <code>Piechart_map</code>, <code>Point_map</code>, and <code>Network_map</code> functions. We note where any updates are function specific.</p>
<ul><li>The maps are now built using the R package <a href="https://cran.r-project.org/web/packages/geodata/geodata.pdf" class="external-link">geodata</a> to avoid the mapping NULL layer error.</li>
<li>Users can now include administrative borders like states using the <code>country_code</code> argument. Country codes must match those used in geodata.</li>
<li>Users can color points by group using the <code>group</code> argument and <code>group_col</code> argument to specify colors for each group (only for <code>Plot_coordinates</code>).</li>
<li>Users can specify the legend position using the <code>legend_pos</code> argument.</li>
<li>Users can add a scale bar or north arrow to plots using the <code>scale_bar</code> and <code>north_arrow</code> arguments. Users can also modify the position and style of the north arrow using <code>north_arrow_position</code> and <code>north_arrow_style</code> arguments, respectively.</li>
<li>Users can add a shapefile to the map with the <code>shapefile</code> argument. Users can also control where the shapefile is placed using <code>shape_file_position</code>, the fill color (<code>shapefile_col</code>), and the outline color (<code>shapefile_outline_col</code>). Shapefiles can be character strings or spatVector objects.</li>
<li>Users can add a raster (discrete or continuous) and control it’s placement in a similar fashion to shapefiles, with the <code>raster</code>, <code>raster_col</code>, and <code>raster_plot_position</code> arguments. Custom breaks can be set with <code>raster_breaks</code> and users can indicate if rasters are discrete using <code>discrete_raster</code>. <strong>Note that sometimes <code>terra</code> interprets continuous rasters as discrete, and users should use the <code>discrete_raster</code> argument to accommodate this.</strong> Rasters can be characters or spatRaster objects.</li>
<li><em>Rasters cannot be added to the `Piechart_map output because it would require additional dependencies to accomodate two fills on the same ggplot2 object. Please email Keaka Farleigh (see package maintainer) if this poses a problem.</em></li>
</ul></div>
<div class="section level3">
<h3 id="pairwise_heatmap-1-4-0"><code>Pairwise_heatmap</code><a class="anchor" aria-label="anchor" href="#pairwise_heatmap-1-4-0"></a></h3>
<ul><li>Users can now specify color gradients of their preferred length (as long as they have that many colors).</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.3.2" id="popgenhelpr-132">PopGenHelpR 1.3.2<a class="anchor" aria-label="anchor" href="#popgenhelpr-132"></a></h2><p class="text-muted">CRAN release: 2024-08-02</p>
<ul><li>July 31st, 2024</li>
<li>Code to calculate Fst in the <code>Differentiation</code> function has been updated to be consistent with the rest of the function (using !is.na).</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.3.1" id="popgenhelpr-131">PopGenHelpR 1.3.1<a class="anchor" aria-label="anchor" href="#popgenhelpr-131"></a></h2><p class="text-muted">CRAN release: 2024-07-26</p>
<ul><li>July 26th, 2024</li>
<li>The previously deprecated <code>Dif_stats</code>, <code>Dif_stats_map</code>, <code>Div_stats</code>, <code>Div_stats_map</code>, and <code>Plot_Ancestry</code> functions have been removed.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.3.0" id="popgenhelpr-130">PopGenHelpR 1.3.0<a class="anchor" aria-label="anchor" href="#popgenhelpr-130"></a></h2><p class="text-muted">CRAN release: 2024-05-09</p>
<ul><li>May 7th, 2024</li>
<li>The <code>Longitude_col</code> and <code>Latitude_col</code> arguments have been added to mapping functions; <code>Plot_coordinates</code>, <code>Ancestry_Piemap</code>, <code>Piechart_map</code>, <code>Network_map</code>, and <code>Point_map</code>.</li>
<li>
<code>Ind_order</code> and <code>Pop_order</code> arguments have been added to <code>Ancestry_Barchart</code> to allow users to specify the order of individuals and populations in the barchart.</li>
<li>The <code>Plot_coordinates</code> function has been added to make sample maps from coordinates.</li>
<li>The <code>Point_map</code> function has replaced <code>Div_stats_map</code> and a statistic argument has been added to highlight the functions utility and allow users to name the map legend.</li>
<li>
<code>Dif_stats_map</code> has been changed to <code>Network_map</code> and a statistic argument has been added to highlight the functions utility and allow users to name the map legend.</li>
<li>
<code>Plot_ancestry</code> has been split into <code>Piechart_Map</code> and <code>Ancestry_barchart</code> so that it is easier for users to determine which function is most appropriate for their analysis.</li>
<li>
<code>Differentiation</code> has been added to estimate Fst, Nei’s D, and Jost’s D. Please see the documentation for more details.</li>
<li>
<code>Heterozygosity</code> has been added to estimate 7 different measures of heterozygosity. Please see the documentation for more details.</li>
<li>
<code>Private.alleles</code> has been added to calculate the number of private alleles in each population.</li>
<li>The <code>Dif_Stats</code> function has been deprecated, please used the <code>Differentiation</code> function to calculate pairwise differentiation between populations (Fst, Nei’s D, Jost’s D) or individuals (Nei’s D).</li>
<li>The <code>Div_Stats</code> function has been deprecated, please use the <code>Heterozygosity</code> function if you wish to estimate heterozygosity or the <code>Private.alleles</code> function if you wish to calculate the number of private alleles per population. Please use the <code>Point_Map</code> function if you wish to visualize the results on a map or plot.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.2.2" id="popgenhelpr-122">PopGenHelpR 1.2.2<a class="anchor" aria-label="anchor" href="#popgenhelpr-122"></a></h2>
<ul><li>October 2nd, 2023</li>
<li>
<code>Dif_stats_Map</code>, <code>Div_Stats_Map</code>, and <code>Plot_ancestry</code> have been updated to use a base color of #f4f4f4 instead of grey99 which was throwing an error for some users.</li>
<li>The piesize argument has been added to <code>Plot_ancestry</code>, the original value of 0.35 was found to be too high, especially in cases where users were mapping a smaller geographic area.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.2.1" id="popgenhelpr-121">PopGenHelpR 1.2.1<a class="anchor" aria-label="anchor" href="#popgenhelpr-121"></a></h2><p class="text-muted">CRAN release: 2023-08-16</p>
<ul><li>August 14th, 2023</li>
<li>
<code>Div_Stats</code> and <code>Dif_stats</code> have been updated to accept a vcf file or vcfR object as input.</li>
<li>
<code>Div_Stats</code> and <code>Dif_stats</code> have been updated to accept a csv file or data frame for population assignment.</li>
<li>
<code>Plot_ancestry</code> has been updated to generate structure-like plots using ggplot2 instead of base R and to handle character and numeric values for individual and population names. Note that individual and populations must be of the same type (i.e., both numeric or both characters).</li>
<li>The dependency <code>rnaturalearth</code> is no longer used. We now use <code>spData</code> for mapping data.</li>
<li>The vignette has been updated to accommodate the changes noted above.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.1.1" id="popgenhelpr-111">PopGenHelpR 1.1.1<a class="anchor" aria-label="anchor" href="#popgenhelpr-111"></a></h2>
<ul><li>July 17th, 2023</li>
<li>The horned lizard data was added so that examples can be run by users.</li>
<li>The write argument has been added to <code>Div_Stats</code> and <code>Dif_stats</code> so that files are not automatically written to the working directory.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.0.1" id="popgenhelpr-101">PopGenHelpR 1.0.1<a class="anchor" aria-label="anchor" href="#popgenhelpr-101"></a></h2>
<ul><li>July 17th, 2023</li>
<li>PopGenHelpR has been updated to rnaturalearthhires in Suggests field of the DESCRIPTION file and now use it conditonally.</li>
</ul></div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.0.0" id="popgenhelpr-100">PopGenHelpR 1.0.0<a class="anchor" aria-label="anchor" href="#popgenhelpr-100"></a></h2><p class="text-muted">CRAN release: 2023-02-13</p>
<ul><li>First development of PopGenHelpR, publication on Github, and submission to the CRAN (02/06/2023)</li>
</ul></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Keaka Farleigh, Mason Murphy, Christopher Blair, Tereza Jezkova.</p>
</div>

<div class="pkgdown-footer-right">
  <p>PopGenHelpR is funded by an National Science Foundation (NSF) Graduate Research Fellowship and NSF Postdoctoral Research Fellowhip in Biology to KF (GRF, Award 2037786; PRFB, DBI-2409958), an NSF grant to CB (DEB-1929679), and an NSF career grant to TJ (DEB-2144047).</p>
</div>

    </footer></div>





  </body></html>

