<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PopGenHelpR">
<title>PopGenHelpR Vignette â€¢ PopGenHelpR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="PopGenHelpR Vignette">
<meta property="og:description" content="PopGenHelpR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PopGenHelpR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Manual</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/PopGenHelpR_vignette.html">Vignette</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_benchmarking.html">Benchmarking</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_heterozygosity.html">Choosing heterozygosity</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_createQmatrix.html">Creating a Q-matrix</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/kfarleigh/PopGenHelpR">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>PopGenHelpR Vignette</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kfarleigh/PopGenHelpR/blob/HEAD/vignettes/PopGenHelpR_vignette.Rmd" class="external-link"><code>vignettes/PopGenHelpR_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>PopGenHelpR_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="welcome">Welcome<a class="anchor" aria-label="anchor" href="#welcome"></a>
</h3>
<p>Welcome to the PopGenHelpR vignette, please contact the authors if
you have any questions about the package. You can also visit our Github
for additional examples (<a href="https://kfarleigh.github.io/PopGenHelpR/" class="uri">https://kfarleigh.github.io/PopGenHelpR/</a>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://kfarleigh.github.io/PopGenHelpR/">PopGenHelpR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview-of-popgenhelpr">Overview of PopGenHelpR<a class="anchor" aria-label="anchor" href="#overview-of-popgenhelpr"></a>
</h2>
<p><code>PopGenHelpR</code> is a one-stop package for data analysis and
visualization. <code>PopGenHelpR</code> can calculate commonly used
population genomic statistics such as heterozygosity and genetic
differentiation, with the functions <code>Heterozygosity</code>,
<code>Differentiation</code>, and <code>Private.alleles</code>. While
also producing publication-quality figures using the functions
<code>Ancestry_barchart</code>, <code>Network_map</code>,
<code>Pairwise_heatmap</code>, and <code>Piechart_map</code>. Check out
the vignette below to see all of these functions in action!</p>
<p><img src="img/PGH_workflow.png" width="750px" height="750px" style="display: block; margin: auto;">
Fig 1. A visualization of the <code>PopGenHelpR</code> workflow.</p>
<div class="section level4">
<h4 id="assumptions-of-popgenhelpr">Assumptions of <code>PopGenHelpR</code><a class="anchor" aria-label="anchor" href="#assumptions-of-popgenhelpr"></a>
</h4>
<p><code>PopGenHelpR</code> is designed to be easy to use, but this also
means that you need to make sure that your data is in order before
analysis and that you pay attention to any warnings output by the
functions.</p>
<ul>
<li>
<strong>Data is assumed to be bi-allelic</strong>. Please reach out
if you have any questions or need help filtering data.</li>
</ul>
<div class="section level5">
<h5 id="load-the-data">Load the data<a class="anchor" aria-label="anchor" href="#load-the-data"></a>
</h5>
<p>First, we will load the data. These data objects are examples of data
types that can be used in the functions in <code>PopGenHelpR</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Fst_dat"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Het_dat"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Q_dat"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HornedLizard_Pop"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HornedLizard_VCF"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="genomic-analysis">Genomic Analysis<a class="anchor" aria-label="anchor" href="#genomic-analysis"></a>
</h3>
<p>Statistical analysis is a critical component population genomics
study, but many R packages only calculate a subset of commonly used
population genomic statistics. <code>PopGenHelpR</code> seeks to remedy
this by allowing researchers to calculate widely used diversity and
differentiation measures in a single package.</p>
<div class="section level4">
<h4 id="heterozygosity">Heterozygosity<a class="anchor" aria-label="anchor" href="#heterozygosity"></a>
</h4>
<p>Heterozygosity is a fundamental statistic in population genomics that
allows researchers to evaluate the genetic diversity of individuals and
populations. <code>PopGenHelpR</code> can estimate 7 measures of
heterozygosity (individual and population). Here we will calculate
observed heterozygosity, but please see the documentation for
<code>Heterozygosity</code> to see all of the options. Better yet, check
out our <a href="https://kfarleigh.github.io/PopGenHelpR/articles/PopGenHelpR_heterozygosity.html">article</a>
on heterozygosity and when to use each measure!</p>
<p>All we need is a vcf or geno file, population assignment file, and
the statistic you wish to estimate (<code>PopGenHelpR</code> does them
all by default). Note that <code>PopGenHelpR</code> assumes that the
first column indicates sample names and the second column indicates the
population to which each individual is assigned. You can use the
arguments <code>individual_col</code> and <code>population_col</code> to
specify which column indicates the sample and population names,
respectively. You can also write the results to a csv if you set
<code>write = TRUE</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Obs_Het</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Ho"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="differentiaton">Differentiaton<a class="anchor" aria-label="anchor" href="#differentiaton"></a>
</h4>
<p>Differentiation is another basic analysis in population genomic
studies. <code>PopGenHelpR</code> allows you to estimate
<em>F<sub>st</sub></em>, <em>Neiâ€™s D</em> (individual and population),
and <em>Jostâ€™s D</em>. Like <code>Heterozygosity</code>, all we need is
a vcf or geno file, population assignment file, and the statistic you
want to calculate (<code>PopGenHelpR</code> does them all by default).
Again, individual and population columns are assumed to be the first and
second columns but can be indicated by users with
<code>individual_col</code> and <code>population_col</code>,
respectively.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Differentiation.html">Differentiation</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Fst"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="private-alleles">Private alleles<a class="anchor" aria-label="anchor" href="#private-alleles"></a>
</h4>
<p>Finally, we will calculate the number of private alleles in each
population. This analysis is often used to evaluate signals of range
expansion and helps researchers identify populations that harbor unique
alleles. Note that <code>Private.alleles</code> can only use a vcf (no
geno files) and does not require you to specify a statistic (all you
absolutely need is a vcf and population file). Otherwise, it operates
just like <code>Heterozygosity</code> or
<code>Differentiation</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Private.alleles.html">Private.alleles</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span><span class="op">)</span></span></code></pre></div>
<p>Letâ€™s move onto visualizations (the fun part), so you can get your
work submitted!</p>
</div>
</div>
<div class="section level3">
<h3 id="visualizations">Visualizations<a class="anchor" aria-label="anchor" href="#visualizations"></a>
</h3>
<p>A strength of <code>PopGenHelpR</code> is its ability to generate
publication-quality figures. You can generate commonly used figures such
as ancestry plots (bar charts and piechart maps), sample maps, and other
figures such as the <code>Network_map</code> that visualizes
relationships between points on a map.</p>
<div class="section level4">
<h4 id="ancestry-plots">Ancestry Plots<a class="anchor" aria-label="anchor" href="#ancestry-plots"></a>
</h4>
<p><code>PopGenHelpR</code> can generate commonly used ancestry
visualizations such as structure-like plots and ancestry piechart maps.
First, we will create structure-like plots for individuals and
populations. We need a q-matrix, population assignments for each
individual, and the number of genetic clusters (K). The q-matrix
represents the contribution of each cluster (K) to an individual or
population and can be obtained from programs like STRUCTURE, ADMIXTURE,
or sNMF. Please see our article on how to extract the q-matrix from
these programs or email Keaka Farleigh.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># First, we seperate the list elements into two seperate objects. The q-matrix (Qmat) and the locality information for each individual (Loc).</span></span>
<span><span class="va">Qmat</span> <span class="op">&lt;-</span> <span class="va">Q_dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">Loc</span> <span class="op">&lt;-</span> <span class="va">Q_dat</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Now we will generate both population and individual plots by setting plot.type to 'all'. If you wanted, you could only generate individual or population plots by setting plot.type to "individual" and "population", respectively.</span></span>
<span><span class="va">Test_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Ancestry_barchart.html">Ancestry_barchart</a></span><span class="op">(</span>anc.mat <span class="op">=</span> <span class="va">Qmat</span>, pops <span class="op">=</span> <span class="va">Loc</span>, K <span class="op">=</span> <span class="fl">5</span>,</span>
<span>plot.type <span class="op">=</span> <span class="st">'all'</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'#d73027'</span>, <span class="st">'#f46d43'</span>, <span class="st">'#e0f3f8'</span>, <span class="st">'#74add1'</span>, <span class="st">'#313695'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Test_all</span><span class="op">$</span><span class="va">`Individual Ancestry Plot`</span></span></code></pre></div>
<p><img src="img/Anc_barchart1.png" width="600px" height="350px" style="display: block; margin: auto;"></p>
<p>We can also generate an ancestry matrix by populations. The ancestry
of each population is calculate by averaging the ancestry of the
individuals in a particular population.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Test_all</span><span class="op">$</span><span class="va">`Population Ancestry Plot`</span></span></code></pre></div>
<p><img src="img/Anc_barchart2.png" width="600px" height="350px" style="display: block; margin: auto;"></p>
<p>Now, we will generate piechart maps of ancestry using the
<code>Piechart_map</code> function. <code>Piechart_map</code> requires
the same input as <code>Ancestry_barchart</code> with the additional
requirement of coordinates for each individual/population. Youâ€™ll notice
that the individual map looks weird; the pie charts have a bunch of
partitions. Thatâ€™s because we have multiple individuals at the same
location, so the population map is probably a better choice. Instead of
layering individuals on top of each other, the population map averages
the ancestry of individuals in a population before mapping. See our
GitHub for additional examples (<a href="https://kfarleigh.github.io/PopGenHelpR/" class="uri">https://kfarleigh.github.io/PopGenHelpR/</a>).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># First, we seperate the list elements into two seperate objects. The q-matrix (Qmat) and the locality information for each individual (Loc).</span></span>
<span><span class="va">Qmat</span> <span class="op">&lt;-</span> <span class="va">Q_dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">Loc</span> <span class="op">&lt;-</span> <span class="va">Q_dat</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># Now we will generate both population and individual plots by setting plot.type to 'all'. If you wanted, you could only generate individual or population plots by setting plot.type to "individual" and "population", respectively.</span></span>
<span><span class="va">Test_all_piemap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Piechart_map.html">Piechart_map</a></span><span class="op">(</span>anc.mat <span class="op">=</span> <span class="va">Qmat</span>, pops <span class="op">=</span> <span class="va">Loc</span>, K <span class="op">=</span> <span class="fl">5</span>,plot.type <span class="op">=</span> <span class="st">'all'</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'#d73027'</span>, <span class="st">'#f46d43'</span>, <span class="st">'#e0f3f8'</span>, <span class="st">'#74add1'</span>, <span class="st">'#313695'</span><span class="op">)</span>, </span>
<span>                                Lat_buffer <span class="op">=</span> <span class="fl">1</span>, Long_buffer <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">Test_all_piemap</span><span class="op">$</span><span class="va">`Individual Map`</span></span></code></pre></div>
<p><img src="img/Ind_PieMap.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
<p>Notice the weird partitions? We can take care of those using the
population piechart map.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Test_all_piemap</span><span class="op">$</span><span class="va">`Population Map`</span></span></code></pre></div>
<p><img src="img/Pop_PieMap.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="differentiation-visualizations">Differentiation visualizations<a class="anchor" aria-label="anchor" href="#differentiation-visualizations"></a>
</h4>
<p><code>PopGenHelpR</code> can use symmetric matrices such as those
output by the <code>Differentiation</code> function to plot heatmaps and
network maps. These plots can be great for understanding the
relationships between populations or individuals.</p>
<p>First, we will use the <code>Pairwise_heatmap</code> function, which
allows us to see relationships between populations or individuals and
only requires a symmetric matrix and legend label (statistic argument).
You can also supply a color vector like we do below, but it is not
required.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PW_hmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Pairwise_heatmap.html">Pairwise_heatmap</a></span><span class="op">(</span><span class="va">Fst_dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, statistic <span class="op">=</span> <span class="st">"Fst"</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#0000FF"</span>, <span class="st">"#FF0000"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="img/PWhmap.png" width="600px" height="600px" style="display: block; margin: auto;"></p>
<p>We can also visualize these relationships on a map using the
<code>Network_map</code> function. This function allows us to visualize
pairwise relationships as the color of links between the points. You
must supply a symmetric matrix (dat argument) and population assignment
file (pops argument). The remaining arguments are optional, but they
allow for greater customization. The neighbors argument, for example,
tells the function how many relationships to visualize, and you can also
use it to specify relationships that you want to see. Please see the
documentation for details.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NW_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Network_map.html">Network_map</a></span><span class="op">(</span><span class="va">Fst_dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, pops <span class="op">=</span> <span class="va">Fst_dat</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, neighbors <span class="op">=</span> <span class="fl">2</span>, statistic <span class="op">=</span> <span class="st">"Fst"</span><span class="op">)</span></span>
<span><span class="va">NW_map</span><span class="op">$</span><span class="va">Map</span></span></code></pre></div>
<p><img src="img/NW_map1.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
<p><code>Network_map</code> can also be used to plot specific
relationships. Letâ€™s isolate the populations with the highest and lowest
Fst by supplying a character vector to the neighbors argument.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NW_map2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Network_map.html">Network_map</a></span><span class="op">(</span><span class="va">Fst_dat</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, pops <span class="op">=</span> <span class="va">Fst_dat</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span>, neighbors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"East_West"</span>, <span class="st">"East_South"</span><span class="op">)</span>, statistic <span class="op">=</span> <span class="st">"Fst"</span><span class="op">)</span></span>
<span><span class="va">NW_map2</span><span class="op">$</span><span class="va">Map</span></span></code></pre></div>
<p><img src="img/NW_map2.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="heterozygosity-and-other-visualizations">Heterozygosity and Other Visualizations<a class="anchor" aria-label="anchor" href="#heterozygosity-and-other-visualizations"></a>
</h4>
<p><code>PopGenHelpR</code> can create maps using output from
<code>Heterozygosity</code> or csv files from external programs to
understand how diversity (or other statistics) is distributed across
geographic space.</p>
<p>We will plot some observed heterozygosity data with the function
<code>Point_map</code>. All you need is a data frame (or csv), and the
name of whatever statistic you are plotting (statistic argument).
<code>Point_map</code> also assumes that the coordinate column names are
Latitude and Longitude.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Het_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Point_map.html">Point_map</a></span><span class="op">(</span><span class="va">Het_dat</span>, statistic <span class="op">=</span> <span class="st">"Heterozygosity"</span><span class="op">)</span></span>
<span><span class="va">Het_map</span><span class="op">$</span><span class="va">`Heterozygosity Map`</span></span></code></pre></div>
<p><img src="img/Het_map1.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
<p>We can also outline the points by setting the <code>out.col</code>
argument.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Het_map2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Point_map.html">Point_map</a></span><span class="op">(</span><span class="va">Het_dat</span>, statistic <span class="op">=</span> <span class="st">"Heterozygosity"</span>, out.col <span class="op">=</span> <span class="st">"#000000"</span><span class="op">)</span></span>
<span><span class="va">Het_map2</span><span class="op">$</span><span class="va">`Heterozygosity Map`</span></span></code></pre></div>
<p><img src="img/Het_map2.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
<p>Finally, we can just plot coordinates using
<code>Plot_coordinates</code>. All we need is a data frame or csv file
with the coordinates for each row indicated by columns names Latitude
and Longitude. You can change the size of the points with the
<code>size</code> argument.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Sample_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Plot_coordinates.html">Plot_coordinates</a></span><span class="op">(</span><span class="va">HornedLizard_Pop</span><span class="op">)</span></span>
<span><span class="va">Sample_map</span></span></code></pre></div>
<p><img src="img/Samp_map1.png" width="500px" height="750px" style="display: block; margin: auto;"></p>
<p>Thank you for your interest in our package, please reach out to Keaka
Farleigh (<a href="mailto:farleik@miamioh.edu" class="email">farleik@miamioh.edu</a>) with any questions, things that
should be included in future versions of the package, or if you would
like to be kept up to date with <code>PopGenHelpR</code>.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Keaka Farleigh, Mason Murphy, Christopher Blair, Tereza Jezkova.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>PopGenHelpR is funded by an National Science Foundation (NSF) Graduate Research Fellowship to KF (Award 2037786), an NSF grant to CB (DEB-1929679), and an NSF career grant to TJ (DEB-2144047).</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
