<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Benchmarking PopGenHelpR with adegenet, hierfstat, mmod, and StAMPP • PopGenHelpR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Benchmarking PopGenHelpR with adegenet, hierfstat, mmod, and StAMPP">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">PopGenHelpR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Manual</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/PopGenHelpR_vignette.html">Vignette</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_benchmarking.html">Benchmarking</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_heterozygosity.html">Choosing heterozygosity</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_PCA.html">Principal Component Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_createQmatrix.html">Creating a Q-matrix</a></li>
    <li><a class="dropdown-item" href="../articles/PopGenHelpR_sortQmatrix.html">Sorting a Q-matrix</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/kfarleigh/PopGenHelpR"><span class="fa fa-github"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Benchmarking PopGenHelpR with adegenet, hierfstat, mmod, and StAMPP</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kfarleigh/PopGenHelpR/blob/main/vignettes/articles/PopGenHelpR_benchmarking.Rmd" class="external-link"><code>vignettes/articles/PopGenHelpR_benchmarking.Rmd</code></a></small>
      <div class="d-none name"><code>PopGenHelpR_benchmarking.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h2>
<p>We will compare the performance of <code>PopGenHelpR</code> with
other R packages available on CRAN. Below, we list the packages we will
compare <code>PopGenHelpR</code> with and the statistics in each
comparison.</p>
<p><strong>F<sub>st</sub> and Nei’s D</strong></p>
<ul>
<li>
<a href="https://cran.r-project.org/web/packages/StAMPP/index.html" class="external-link">StAMPP</a>
(<a href="https://doi.org/10.1111/1755-0998.12129" class="external-link">Pembleton et al.,
2013</a>)</li>
</ul>
<p><strong>Jost’s D</strong></p>
<ul>
<li>
<a href="https://cran.r-project.org/web/packages/mmod/index.html" class="external-link">mmod</a>
(Winter et al., 2017)</li>
</ul>
<p><strong>Expected and Observed Heterozygosity</strong></p>
<ul>
<li><p><a href="https://cran.r-project.org/web/packages/hierfstat/index.html" class="external-link">hierfstat</a>
(<a href="https://doi.org/10.1111/j.1471-8286.2004.00828.x" class="external-link">Goudet,
2005</a>)</p></li>
<li><p><a href="https://cran.r-project.org/web/packages/adegenet/index.html" class="external-link">adegenet</a>
(<a href="https://doi.org/10.1093/bioinformatics/btn129" class="external-link">Jombart,
2008</a>)</p></li>
</ul>
<div class="section level3">
<h3 id="lets-begin">Let’s Begin<a class="anchor" aria-label="anchor" href="#lets-begin"></a>
</h3>
<p>First we will load the packages and the data in
<code>PopGenHelpR</code>. The data comes from <a href="https://doi.org/10.1111/mec.16070" class="external-link">Farleigh et al. (2021)</a>. We
also load <code>vcfR</code> to convert between data formats (<a href="https://doi.org/10.1111/1755-0998.12549" class="external-link">Knaus &amp; Grunwald,
2017</a>).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install packages if needed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/thibautjombart/adegenet" class="external-link">"adegenet"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"adegenet"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    /// adegenet 2.1.10 is loaded ////////////</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    &gt; overview: '?adegenet'</span></span>
<span><span class="co">#&gt;    &gt; tutorials/doc/questions: 'adegenetWeb()' </span></span>
<span><span class="co">#&gt;    &gt; bug reports/feature requests: adegenetIssues()</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://www.r-project.org" class="external-link">"hierfstat"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"hierfstat"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'hierfstat'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:adegenet':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Hs, read.fstat</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lpembleton/StAMPP" class="external-link">"StAMPP"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"StAMPP"</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'ape'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:hierfstat':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     pcoa, varcomp</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'pegas':</span></span>
<span><span class="co">#&gt;   method      from</span></span>
<span><span class="co">#&gt;   print.amova ade4</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'pegas'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ape':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     mst</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:ade4':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     amova</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://github.com/dwinter/mmod" class="external-link">"mmod"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"mmod"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load the packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://kfarleigh.github.io/PopGenHelpR/">PopGenHelpR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thibautjombart/adegenet" class="external-link">adegenet</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.r-project.org" class="external-link">hierfstat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lpembleton/StAMPP" class="external-link">StAMPP</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/dwinter/mmod" class="external-link">mmod</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/knausb/vcfR" class="external-link">vcfR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;    *****       ***   vcfR   ***       *****</span></span>
<span><span class="co">#&gt;    This is vcfR 1.15.0 </span></span>
<span><span class="co">#&gt;      browseVignettes('vcfR') # Documentation</span></span>
<span><span class="co">#&gt;      citation('vcfR') # Citation</span></span>
<span><span class="co">#&gt;    *****       *****      *****       *****</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'vcfR'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:pegas':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     getINFO, write.vcf</span></span>
<span></span>
<span><span class="co"># Load the data </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HornedLizard_VCF"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HornedLizard_Pop"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="fst-and-neis-d-comparison">
<em>F<sub>ST</sub></em> and <em>Nei’s D</em> Comparison<a class="anchor" aria-label="anchor" href="#fst-and-neis-d-comparison"></a>
</h2>
<p>We will compare <code>PopGenHelpR</code> and <code>StAMPP</code>.
Both packages use the formulas from Weir and Cockerham (1984) ane Nei
(1972) to calculate <em>F<sub>ST</sub></em> and <em>Nei’s D</em>,
respectively but we want to make sure that our estimates are consistent
across packages.</p>
<p>First, we need to format the data for <code>StAMPP</code></p>
<p>Now we can calculate our statistics. Let’s start with
<em>F<sub>ST</sub></em>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PGH_fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Differentiation.html">Differentiation</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Fst"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "vcfR object detected, proceeding to formatting."</span></span>
<span><span class="co">#&gt; Formatting has finished, moving onto calculations</span></span>
<span></span>
<span><span class="va">Stmp_fst</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/StAMPP/man/stamppFst.html" class="external-link">stamppFst</a></span><span class="op">(</span><span class="va">Glight</span>, nboots <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<p>Let’s inspect the results.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PGH_fst</span><span class="op">$</span><span class="va">Fst</span></span>
<span><span class="co">#&gt;            East     South West</span></span>
<span><span class="co">#&gt; East         NA        NA   NA</span></span>
<span><span class="co">#&gt; South 0.2511135        NA   NA</span></span>
<span><span class="co">#&gt; West  0.3905512 0.3029886   NA</span></span>
<span><span class="va">Stmp_fst</span></span>
<span><span class="co">#&gt;            East     South West</span></span>
<span><span class="co">#&gt; East         NA        NA   NA</span></span>
<span><span class="co">#&gt; South 0.2511135        NA   NA</span></span>
<span><span class="co">#&gt; West  0.3905512 0.3029886   NA</span></span>
<span></span>
<span><span class="co"># Is there a difference between the two?</span></span>
<span><span class="va">Fst_comparison</span> <span class="op">&lt;-</span> <span class="va">PGH_fst</span><span class="op">$</span><span class="va">Fst</span><span class="op">-</span><span class="va">Stmp_fst</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">Fst_comparison</span><span class="op">)</span></span>
<span><span class="co">#&gt;       East       South        West    </span></span>
<span><span class="co">#&gt;  Min.   :0   Min.   :0   Min.   : NA  </span></span>
<span><span class="co">#&gt;  1st Qu.:0   1st Qu.:0   1st Qu.: NA  </span></span>
<span><span class="co">#&gt;  Median :0   Median :0   Median : NA  </span></span>
<span><span class="co">#&gt;  Mean   :0   Mean   :0   Mean   :NaN  </span></span>
<span><span class="co">#&gt;  3rd Qu.:0   3rd Qu.:0   3rd Qu.: NA  </span></span>
<span><span class="co">#&gt;  Max.   :0   Max.   :0   Max.   : NA  </span></span>
<span><span class="co">#&gt;  NA's   :1   NA's   :2   NA's   :3</span></span></code></pre></div>
<p>Now we move onto <em>Nei’s D</em>. We can use the same genlight that
we created for the <em>F<sub>ST</sub></em> calculations. We will
calculate <em>Nei’s D</em> for both population’s and individual’s.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PGH_ND</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Differentiation.html">Differentiation</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"NeisD"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "vcfR object detected, proceeding to formatting."</span></span>
<span><span class="co">#&gt; Formatting has finished, moving onto calculations</span></span>
<span></span>
<span><span class="co"># StAMPP population Nei's D</span></span>
<span><span class="va">Stmp_popND</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/StAMPP/man/stamppNeisD.html" class="external-link">stamppNeisD</a></span><span class="op">(</span><span class="va">Glight</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># StAMPP individual Nei's D</span></span>
<span><span class="va">Stmp_indND</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/StAMPP/man/stamppNeisD.html" class="external-link">stamppNeisD</a></span><span class="op">(</span><span class="va">Glight</span>, pop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Compare the results like we did with <em>F<sub>ST</sub></em>. Note
that <code>PopGenHelpR</code> only reports result on the lower
triangular element so we will set the upper triangular element of the
<code>Stmp_popND</code> and <code>Stmp_indND</code> objects to NA.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Population comparison</span></span>
<span><span class="va">PGH_ND</span><span class="op">$</span><span class="va">NeisD_pop</span></span>
<span><span class="co">#&gt;             East     South West</span></span>
<span><span class="co">#&gt; East  0.00000000        NA   NA</span></span>
<span><span class="co">#&gt; South 0.09005846 0.0000000   NA</span></span>
<span><span class="co">#&gt; West  0.19806009 0.1148848    0</span></span>
<span><span class="va">Stmp_popND</span></span>
<span><span class="co">#&gt;           [,1]     [,2]     [,3]</span></span>
<span><span class="co">#&gt; East  0.000000 0.090058 0.198060</span></span>
<span><span class="co">#&gt; South 0.090058 0.000000 0.114885</span></span>
<span><span class="co">#&gt; West  0.198060 0.114885 0.000000</span></span>
<span></span>
<span><span class="co"># Set StAMPP upper diagnoals to NA</span></span>
<span><span class="va">Stmp_popND</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html" class="external-link">upper.tri</a></span><span class="op">(</span><span class="va">Stmp_popND</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span><span class="va">Stmp_indND</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lower.tri.html" class="external-link">upper.tri</a></span><span class="op">(</span><span class="va">Stmp_indND</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NA</span></span>
<span></span>
<span><span class="va">popND_comparison</span> <span class="op">&lt;-</span> <span class="va">PGH_ND</span><span class="op">$</span><span class="va">NeisD_pop</span><span class="op">-</span><span class="va">Stmp_popND</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">popND_comparison</span><span class="op">)</span></span>
<span><span class="co">#&gt;       East               South             West  </span></span>
<span><span class="co">#&gt;  Min.   :0.000e+00   Min.   :-2e-07   Min.   :0  </span></span>
<span><span class="co">#&gt;  1st Qu.:4.518e-08   1st Qu.:-1e-07   1st Qu.:0  </span></span>
<span><span class="co">#&gt;  Median :9.036e-08   Median :-1e-07   Median :0  </span></span>
<span><span class="co">#&gt;  Mean   :1.840e-07   Mean   :-1e-07   Mean   :0  </span></span>
<span><span class="co">#&gt;  3rd Qu.:2.760e-07   3rd Qu.: 0e+00   3rd Qu.:0  </span></span>
<span><span class="co">#&gt;  Max.   :4.615e-07   Max.   : 0e+00   Max.   :0  </span></span>
<span><span class="co">#&gt;                      NA's   :1        NA's   :2</span></span>
<span></span>
<span><span class="co"># Get the mean difference</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">popND_comparison</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 6.038476e-08</span></span>
<span></span>
<span><span class="co"># Individual comparison, uncomment if you want to see it</span></span>
<span><span class="co">#PGH_ND$NeisD_ind</span></span>
<span><span class="co">#Stmp_indND</span></span>
<span></span>
<span><span class="va">indND_comparison</span> <span class="op">&lt;-</span> <span class="va">PGH_ND</span><span class="op">$</span><span class="va">NeisD_ind</span> <span class="op">-</span> <span class="va">Stmp_indND</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">indND_comparison</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 5.807753e-09</span></span></code></pre></div>
<p>We see that the difference is very small and is because of rounding.
Let’s move onto <em>Jost’s D</em> comparison with <code>mmod</code>.</p>
</div>
<div class="section level2">
<h2 id="josts-d-comparison">
<em>Jost’s D</em> Comparison<a class="anchor" aria-label="anchor" href="#josts-d-comparison"></a>
</h2>
<p>We will compare <code>PopGenHelpR</code> and <code>mmod</code>. Both
packages use the formulas from Jost (2008). <code>mmod</code> uses
genind objects so we have to do some format conversion first.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Genind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/vcfR/man/vcfR_conversion.html" class="external-link">vcfR2genind</a></span><span class="op">(</span><span class="va">HornedLizard_VCF</span><span class="op">)</span></span>
<span><span class="va">Genind</span><span class="op">@</span><span class="va">pop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="va">HornedLizard_Pop</span><span class="op">$</span><span class="va">Population</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/adegenet/man/accessors.html" class="external-link">ploidy</a></span><span class="op">(</span><span class="va">Genind</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span></span>
<span><span class="co"># Calculate Jost's D</span></span>
<span><span class="va">PGH_JD</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Differentiation.html">Differentiation</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"JostsD"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "vcfR object detected, proceeding to formatting."</span></span>
<span><span class="co">#&gt; Formatting has finished, moving onto calculations</span></span>
<span><span class="va">mmod_JD</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mmod/man/pairwise_D.html" class="external-link">pairwise_D</a></span><span class="op">(</span><span class="va">Genind</span><span class="op">)</span></span>
<span></span>
<span><span class="va">PGH_JD</span><span class="op">$</span><span class="va">JostsD</span></span>
<span><span class="co">#&gt;             East    South West</span></span>
<span><span class="co">#&gt; East  0.00000000       NA   NA</span></span>
<span><span class="co">#&gt; South 0.08135043 0.000000   NA</span></span>
<span><span class="co">#&gt; West  0.17491621 0.103915    0</span></span>
<span><span class="va">mmod_JD</span></span>
<span><span class="co">#&gt;             East      South</span></span>
<span><span class="co">#&gt; South 0.08135043           </span></span>
<span><span class="co">#&gt; West  0.17370519 0.10440251</span></span>
<span></span>
<span><span class="co"># Compare differences mathematically</span></span>
<span><span class="va">PGH_JD</span><span class="op">$</span><span class="va">JostsD</span><span class="op">[</span><span class="fl">2</span><span class="op">:</span><span class="fl">3</span>,<span class="fl">1</span><span class="op">]</span> <span class="op">-</span> <span class="va">mmod_JD</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="co">#&gt;         South          West </span></span>
<span><span class="co">#&gt; -6.938894e-17  1.211019e-03</span></span>
<span><span class="va">PGH_JD</span><span class="op">$</span><span class="va">JostsD</span><span class="op">[</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">-</span> <span class="va">mmod_JD</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] -0.1044025</span></span></code></pre></div>
<p>Estimate’s are very similar between <code>PopGenHelpR</code> and
<code>mmod</code>, we will move onto heterozygosity.</p>
</div>
<div class="section level2">
<h2 id="expected-and-observed-heterozygosity-comparison">
<em>Expected</em> and <em>Observed Heterozygosity</em>
Comparison<a class="anchor" aria-label="anchor" href="#expected-and-observed-heterozygosity-comparison"></a>
</h2>
<p>We will compare <code>PopGenHelpR</code>, <code>hierfstat</code>, and
<code>adegenet</code>. Again, the packages use the same formula’s, so we
expect similar if not identical results. <code>hierfstat</code> uses
it’s own format, so we will convert the data before calculations.
Luckily we can convert the genind object from <em>Jost’s D</em>
comparisons.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Hstat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hierfstat/man/genind2hierfstat.html" class="external-link">genind2hierfstat</a></span><span class="op">(</span><span class="va">Genind</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Calculate heterozygosities</span></span>
<span><span class="co"># Expected</span></span>
<span><span class="va">PGH_He</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"He"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "vcfR object detected, proceeding to formatting."</span></span>
<span><span class="co">#&gt; Formatting has finished, moving onto calculations</span></span>
<span><span class="co"># Observed</span></span>
<span><span class="va">PGH_Ho</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Ho"</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "vcfR object detected, proceeding to formatting."</span></span>
<span><span class="co">#&gt; Formatting has finished, moving onto calculations</span></span>
<span></span>
<span><span class="va">Hstat_hets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hierfstat/man/basic.stats.html" class="external-link">basic.stats</a></span><span class="op">(</span><span class="va">Hstat</span><span class="op">)</span></span>
<span><span class="va">Hstat_Ho</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colMeans</a></span><span class="op">(</span><span class="va">Hstat_hets</span><span class="op">$</span><span class="va">Ho</span><span class="op">)</span></span>
<span></span>
<span><span class="va">He_adnet</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/adegenet/man/Hs.html" class="external-link">Hs</a></span><span class="op">(</span><span class="va">Genind</span><span class="op">)</span></span>
<span></span>
<span><span class="va">PGH_He</span><span class="op">$</span><span class="va">He_perpop</span><span class="op">$</span><span class="va">Expected.Heterozygosity</span><span class="op">-</span><span class="va">He_adnet</span></span>
<span><span class="co">#&gt;         East        South         West </span></span>
<span><span class="co">#&gt; -0.006854206 -0.003143262 -0.006625364</span></span>
<span><span class="va">PGH_Ho</span><span class="op">$</span><span class="va">Ho_perpop</span><span class="op">$</span><span class="va">Observed.Heterozygosity</span><span class="op">-</span><span class="va">Hstat_Ho</span></span>
<span><span class="co">#&gt;          East         South          West </span></span>
<span><span class="co">#&gt;  7.511576e-06 -1.790920e-06  0.000000e+00</span></span></code></pre></div>
<p>We again see very small differences between the estimates.</p>
<p>Please reach out to Keaka Farleigh (<a href="mailto:farleik@miamioh.edu" class="email">farleik@miamioh.edu</a>)
if you have any questions, and please see the references and
acknowledgments below.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Farleigh, K., Vladimirova, S. A., Blair, C., Bracken, J. T.,
Koochekian, N., Schield, D. R., … &amp; Jezkova, T. (2021). The effects
of climate and demographic history in shaping genomic variation across
populations of the Desert Horned Lizard (Phrynosoma platyrhinos).
Molecular Ecology, 30(18), 4481-4496.</p>
<p>Goudet, J. (2005). hierfstat, a package for R to compute and test
hierarchical F‐statistics. Molecular ecology notes, 5(1), 184-186.</p>
<p>Jost, L. (2008). GST and its relatives do not measure
differentiation. Molecular ecology, 17(18), 4015-4026.</p>
<p>Knaus, B. J., &amp; Grünwald, N. J. (2017). vcfr: a package to
manipulate and visualize variant call format data in R. Molecular
ecology resources, 17(1), 44-53.</p>
<p>Nei, M. (1972). Genetic distance between populations. The American
Naturalist, 106(949), 283-292.</p>
<p>Pembleton, L. W., Cogan, N. O., &amp; Forster, J. W. (2013). St AMPP:
An R package for calculation of genetic differentiation and structure of
mixed‐ploidy level populations. Molecular ecology resources, 13(5),
946-952.</p>
<p>Weir, B. S., &amp; Cockerham, C. C. (1984). Estimating F-statistics
for the analysis of population structure. evolution, 1358-1370.</p>
<p>Winter, D., Green, P., Kamvar, Z., &amp; Gosselin, T. (2017). mmod:
modern measures of population differentiation (Version 1.3.3).</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>We thank the authors of <code>hierfstat</code>, <code>mmod</code>,
<code>StAMPP</code>, and all of the package dependencies. They provided
inspiration for <code>PopGenHelpR</code> and their commitment to open
science made it possible to develop and benchmark our package.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Keaka Farleigh, Mason Murphy, Christopher Blair, Tereza Jezkova.</p>
</div>

<div class="pkgdown-footer-right">
  <p>PopGenHelpR is funded by an National Science Foundation (NSF) Graduate Research Fellowship to KF (Award 2037786), an NSF grant to CB (DEB-1929679), and an NSF career grant to TJ (DEB-2144047).</p>
</div>

    </footer>
</div>





  </body>
</html>
