<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PopGenHelpR">
<title>Which heterozygosity should I use? • PopGenHelpR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Which heterozygosity should I use?">
<meta property="og:description" content="PopGenHelpR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PopGenHelpR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Manual</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/PopGenHelpR_vignette.html">Vignette</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_benchmarking.html">Benchmarking</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_heterozygosity.html">Choosing heterozygosity</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_PCA.html">Principal Component Analysis</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_createQmatrix.html">Creating a Q-matrix</a>
    <a class="dropdown-item" href="../articles/PopGenHelpR_sortQmatrix.html">Sorting a Q-matrix</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/kfarleigh/PopGenHelpR">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Which heterozygosity should I use?</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/kfarleigh/PopGenHelpR/blob/HEAD/vignettes/articles/PopGenHelpR_heterozygosity.Rmd" class="external-link"><code>vignettes/articles/PopGenHelpR_heterozygosity.Rmd</code></a></small>
      <div class="d-none name"><code>PopGenHelpR_heterozygosity.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="purpose">Purpose<a class="anchor" aria-label="anchor" href="#purpose"></a>
</h2>
<p>To help you understand the different measures of heterozygosity in
<code>PopGenHelpR</code> and determine which measure is appropriate for
your question/objective.</p>
</div>
<div class="section level2">
<h2 id="what-is-heterozygosity-and-why-is-it-important">What is heterozygosity and why is it important?<a class="anchor" aria-label="anchor" href="#what-is-heterozygosity-and-why-is-it-important"></a>
</h2>
<p>Heterozygosity refers to the presence of two alleles at a locus. We
often use heterozygosity to measure genetic diversity, which is
essential for a species’ ability to adapt and persist.</p>
</div>
<div class="section level2">
<h2 id="what-measures-of-heterozygosity-can-popgenhelpr-estimate">What measures of heterozygosity can <code>PopGenHelpR</code>
estimate?<a class="anchor" aria-label="anchor" href="#what-measures-of-heterozygosity-can-popgenhelpr-estimate"></a>
</h2>
<p><code>PopGenHelpR</code> can estimate seven measures of
heterozygosity with the function <code>Heterozygosity</code>. We list
each measure below before providing brief descriptions of each one.</p>
<ul>
<li>
<strong><em>Population Measures</em></strong>
<ul>
<li>Observed heterozygosity (<em>H<sub>o</sub></em>)</li>
<li>Expected heterozygosity (<em>H<sub>e</sub></em>)</li>
</ul>
</li>
<li>
<strong><em>Individual Measures</em></strong>
<ul>
<li>Proportion of heterozygous loci (<em>PHt</em>)</li>
<li>Proportion of heterozygous loci standardized by the average expected
heterozygosity (<em>Hs<sub>exp</sub></em>)</li>
<li>Proportion of heterozygous loci standardized by the average observed
heterozygosity (<em>Hs<sub>obs</sub></em>)</li>
<li>Internal relatedness (<em>IR</em>)</li>
<li>Homozygosity by locus (<em>HL</em>)</li>
</ul>
</li>
</ul>
<p><code>PopGenHelpR</code> can calculate all of these measures using
the <code>Heterozygosity</code> function. See the code below.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load package and toy data for all of the statistics</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://kfarleigh.github.io/PopGenHelpR/">PopGenHelpR</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HornedLizard_Pop"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HornedLizard_VCF"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">All_Het</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="population-measures-of-heterozygosity">Population measures of heterozygosity<a class="anchor" aria-label="anchor" href="#population-measures-of-heterozygosity"></a>
</h2>
<p><code>PopGenHelpR</code> users can estimate the expected and observed
heterozygosity (<em>H<sub>e</sub></em> and <em>H<sub>o</sub></em>,
respectively) of each population in their data set.</p>
<div class="section level3">
<h3 id="expected-heterozygosity-he">Expected heterozygosity (<em>H<sub>e</sub></em>)<a class="anchor" aria-label="anchor" href="#expected-heterozygosity-he"></a>
</h3>
<p><code>PopGenHelpR</code> estimates <em>H<sub>e</sub></em> per locus
and population following the equations provided by the Hardy-Weinberg
equation. Briefly, the equation estimates <em>H<sub>e</sub></em> as one
minus the squared frequency of each allele (<span class="math inline">\(p^2\)</span> and <span class="math inline">\(q^2\)</span>, respectively), thus giving us the
expected frequency of heterozygous genotypes (2pq) at a locus. The
overall measure of <em>H<sub>e</sub></em> is calculated as the average
of the per locus estimates.</p>
<p>The equation per locus is below, where <em>p</em> is the reference
allele and <em>q</em> is the alternate allele:</p>
<p><span class="math display">\[
H_e = 1-p^2-q^2
\]</span></p>
<p>Thus, the equation to calculate the overall <em>H<sub>e</sub></em> is
below, where <em>K</em> is the number of SNPs.</p>
<p><span class="math display">\[
H_e =  \frac{\sum_{k=1}^K(1-p^2-q^2)}{K}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-he">How do we use <em>H<sub>e</sub></em><a class="anchor" aria-label="anchor" href="#how-do-we-use-he"></a>
</h4>
<p>We use <em>H<sub>e</sub></em> as a null model to test against and
determine if Hardy-Weinberg equilibrium is being violated. Violations
could indicate mutation, non-random mating, gene flow, non-infinite
population size, natural selection, or any combination.</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-he-in-popgenhelpr">How do we calculate <em>H<sub>e</sub></em> in
<code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-he-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>H<sub>e</sub></em> in <code>PopGenHelpR</code>
using the command below.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">He</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"He"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="observed-heterozygosity-ho">Observed heterozygosity (<em>H<sub>o</sub></em>)<a class="anchor" aria-label="anchor" href="#observed-heterozygosity-ho"></a>
</h3>
<p><code>PopGenHelpR</code> estimates <em>H<sub>o</sub></em> per locus
and population following the equations of Nei (1987). Briefly, the
equations estimate <em>H<sub>o</sub></em> as one minus the proportion of
homozygotes in the population at each locus, thus giving us the
proportion of heterozygotes at a locus. The overall measure of
<em>H<sub>o</sub></em> is calculated as the average of the per locus
estimates.</p>
<p>The equation per locus is below:</p>
<p><span class="math display">\[
H_o = 1- \frac{Number\; of\; homoyzgotes}{Number\; of\; samples}
\]</span></p>
<p>Thus the overall measure of <em>H<sub>o</sub></em> is below, where K
is the number of SNPs:</p>
<p><span class="math display">\[
H_o = \frac{\sum_{k = 1}^K{1- \frac{Number\; of\; homoyzgotes}{Number\;
of\; samples}}}{K}
\]</span></p>
<p>The formal equation of <em>H<sub>o</sub></em> from Nei (1987) is
below: <em>Pkii</em> is the proportion of homozygote (<em>i</em>) in a
sample (<em>k</em>), and <em>np</em> is the number of samples:</p>
<p><span class="math display">\[
H_o = 1-\sum_{k}\sum_{i}\frac{Pkii}{np}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-ho">How do we use <em>H<sub>o</sub></em><a class="anchor" aria-label="anchor" href="#how-do-we-use-ho"></a>
</h4>
<p>We use <em>H<sub>o</sub></em> as a measure of genetic diversity and
also to compare to <em>H<sub>e</sub></em> to determine if our data is
exhibiting different patterns, such as inbreeding
(<em>H<sub>o</sub></em> &lt; <em>H<sub>e</sub></em>) or heterozygote
advantage (<em>H<sub>o</sub></em> &gt; <em>H<sub>e</sub></em>).</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-ho-in-popgenhelpr">How do we calculate <em>H<sub>o</sub></em> in
<code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-ho-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>H<sub>o</sub></em> in <code>PopGenHelpR</code>
using the command below.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Ho</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Ho"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="individual-measures-of-heterozygosity">Individual measures of heterozygosity<a class="anchor" aria-label="anchor" href="#individual-measures-of-heterozygosity"></a>
</h2>
<p><code>PopGenHelpR</code> users can estimate the proportion of
heterozygous loci (<em>PHt</em>), the proportion of heterozygous loci
standardized by the average expected heterozygosity
(<em>Hs<sub>exp</sub></em>), the proportion of heterozygous loci
standardized by the average observed heterozygosity
(<em>Hs<sub>obs</sub></em>), the internal relatedness (<em>IR</em>), and
the homozygosity by locus (<em>HL</em>) of individuals in their data
set.</p>
<div class="section level3">
<h3 id="proportion-of-heterozygous-loci-pht">Proportion of heterozygous loci (<em>PHt</em>)<a class="anchor" aria-label="anchor" href="#proportion-of-heterozygous-loci-pht"></a>
</h3>
<p>The proportion of heterozygous loci (<em>PHt</em>) is calculated as
the number of heterozygous SNPs divided by the number of genotyped SNPs
in each individual.</p>
<p><span class="math display">\[
PHt = \frac{Number\; of\; heterozygous\; SNPs}{Number\; of\; genotyped\;
SNPs}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-pht">How do we use <em>PHt</em><a class="anchor" aria-label="anchor" href="#how-do-we-use-pht"></a>
</h4>
<p><em>PHt</em> is helpful for evaluating the diversity within each
individual and comparing it to other samples. Individual heterozygosity
is also commonly used to investigate inbreeding (Miller et al., 2014).
Individual heterozygosity is used in heterozygosity-fitness correlations
(HFC), assuming that heterozygosity positively correlates with fitness.
Thus, increased heterozygosity (decreased inbreeding) indicates higher
fitness.</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-pht-in-popgenhelpr">How do we calculate <em>PHt</em> in <code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-pht-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>PHt</em> in <code>PopGenHelpR</code> using the
command below.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">PHt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"PHt"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="proportion-of-heterozygous-loci-standardized-by-the-average-expected-heterozygosity-hsexp">Proportion of heterozygous loci standardized by the average expected
heterozygosity (<em>Hs<sub>exp</sub></em>)<a class="anchor" aria-label="anchor" href="#proportion-of-heterozygous-loci-standardized-by-the-average-expected-heterozygosity-hsexp"></a>
</h3>
<p>The proportion of heterozygous loci standardized by the average
expected heterozygosity (<em>Hs<sub>exp</sub></em>) is calculated as
<em>PHt</em> divided by the mean expected heterozygosity
(<em>H<sub>e</sub></em>) for each individual. Please see the equation
below.</p>
<p><span class="math display">\[
Hs_{exp} = \frac{PHt}{H_e}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-hsexp">How do we use <em>Hs<sub>exp</sub></em><a class="anchor" aria-label="anchor" href="#how-do-we-use-hsexp"></a>
</h4>
<p><em>Hs<sub>exp</sub></em> was introduced by Coltman et al. (1999) to
evaluate individual heterozygosity across individuals who were genotyped
with different markers; this allows us to compare individual
heterozygosity on the same scale and to assess inbreeding. Like
<em>PHt</em>, higher <em>Hs<sub>exp</sub></em> indicates less
inbreeding.</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-hsexp-in-popgenhelpr">How do we calculate <em>Hs<sub>exp</sub></em> in
<code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-hsexp-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>Hs<sub>exp</sub></em> in
<code>PopGenHelpR</code> using the command below.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Hs_exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Hs_exp"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="proportion-of-heterozygous-loci-standardized-by-the-average-observed-heterozygosity-hsobs">Proportion of heterozygous loci standardized by the average observed
heterozygosity (<em>Hs<sub>obs</sub></em>)<a class="anchor" aria-label="anchor" href="#proportion-of-heterozygous-loci-standardized-by-the-average-observed-heterozygosity-hsobs"></a>
</h3>
<p>The proportion of heterozygous loci standardized by the average
observed heterozygosity (<em>Hs<sub>obs</sub></em>) is calculated as
<em>PHt</em> divided by the mean observed heterozygosity
(<em>H<sub>o</sub></em>) for each individual. Please see the equation
below.</p>
<p><span class="math display">\[
Hs_{obs} = \frac{PHt}{H_o}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-hsobs">How do we use <em>Hs<sub>obs</sub></em><a class="anchor" aria-label="anchor" href="#how-do-we-use-hsobs"></a>
</h4>
<p><em>Hs<sub>obs</sub></em> was introduced by Coltman et al. (1999) to
evaluate individual heterozygosity across individuals who were genotyped
with different markers; this allows us to compare individual
heterozygosity on the same scale and to assess inbreeding. Like
<em>PHt</em>, higher <em>Hs<sub>obs</sub></em> indicates less
inbreeding.</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-hsobs-in-popgenhelpr">How do we calculate <em>Hs<sub>obs</sub></em> in
<code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-hsobs-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>Hs<sub>obs</sub></em> in
<code>PopGenHelpR</code> using the command below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Hs_obs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"Hs_obs"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="internal-relatedness-ir">Internal relatedness (<em>IR</em>)<a class="anchor" aria-label="anchor" href="#internal-relatedness-ir"></a>
</h3>
<p>The equation for Internal relatedness (<em>IR</em>) is more complex
and qutie the mouthful(or sentence full?). Please see the equation
below. <em>IR</em> is calculated as two times the number of homozygous
loci minus the sum of the frequency of the ith allele divided by two
times the number of loci minus the sum of the frequency of the ith
allele (see equation 2.1 in Amos et al., 2001).</p>
<p><span class="math display">\[
IR = \frac{(2H-\sum{f_i})}{(2N-\sum{f_i})}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-ir">How do we use <em>IR</em>?<a class="anchor" aria-label="anchor" href="#how-do-we-use-ir"></a>
</h4>
<p><em>IR</em> was developed by Amos et al. (2001) to measure the
diversity within individuals (Amos et al., 2001). Negative <em>IR</em>
values suggest that individuals are outbred (tend to be more
heterozygous), while positive values indicate that individuals are
inbred (tend to be more homozygous).</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-ir-in-popgenhelpr">How do we calculate <em>IR</em> in <code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-ir-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>IR</em> in <code>PopGenHelpR</code> using the
command below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">IR</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"IR"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="homozygosity-by-locus-hl">Homozygosity by locus (<em>HL</em>)<a class="anchor" aria-label="anchor" href="#homozygosity-by-locus-hl"></a>
</h3>
<p>Homozygosity by locus (<em>HL</em>) is calculated as the expected
heterozygosity of loci in homozygosis (<span class="math inline">\(E_h\)</span>) divided by the sum of the expected
heterozygosity of loci in homozygosis (<span class="math inline">\(E_h\)</span>) and the expected heterozygosity of
loci in heterozygosis (<span class="math inline">\(E_j\)</span>; see
Aparicio et al., 2006). Please see the equation below.</p>
<p><span class="math display">\[
HL = \frac{\sum{E_h}}{\sum{E_h} + \sum{E_j}}
\]</span></p>
<div class="section level4">
<h4 id="how-do-we-use-hl">How do we use <em>HL</em>?<a class="anchor" aria-label="anchor" href="#how-do-we-use-hl"></a>
</h4>
<p><em>HL</em> was proposed by Aparicio et al. (2006) to improve on
<em>IR</em> by weighing the contribution of each locus to the index
depending on their allelic variability (Aparicio et al., 2006).
<em>HL</em>, like <em>IR</em>, is useful for evaluating the diversity
within an individual. <em>HL</em> ranges from 0 when all loci are
heterozygous and 1 when all loci are homozygous (Aparicio et al.,
2006).</p>
</div>
<div class="section level4">
<h4 id="how-do-we-calculate-hl-in-popgenhelpr">How do we calculate <em>HL</em> in <code>PopGenHelpR</code>?<a class="anchor" aria-label="anchor" href="#how-do-we-calculate-hl-in-popgenhelpr"></a>
</h4>
<p>You can calculate <em>HL</em> in <code>PopGenHelpR</code> using the
command below.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HL</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Heterozygosity.html">Heterozygosity</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">HornedLizard_VCF</span>, pops <span class="op">=</span> <span class="va">HornedLizard_Pop</span>, statistic <span class="op">=</span> <span class="st">"HL"</span><span class="op">)</span></span></code></pre></div>
<p>Please reach out to Keaka Farleigh (<a href="mailto:farleik@miamioh.edu" class="email">farleik@miamioh.edu</a>)
if you have questions or need any help.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Amos W., Worthington Wilmer J., Fullard K., Burg T. M., Croxall J.
P., Bloch D., Coulson T. 2001. The influence of parental relatedness on
reproductive success. Proceedings of the Royal Society B: Biological
Sciences. 268: 2021-2027.</p>
<p>Aparicio J. M., Ortego J., Cordero P. J. 2006. What should we weigh
to estimate heterozygosity, alleles or loci? Molecular Ecology. 15:
4659-4665</p>
<p>Coltman D. W., Pilkington J. G., Smith J. A., Pemberton J. M. 1999.
Parasite-mediated selection against inbred Soay sheep in a free-living,
island population. Evolution. 53: 1259-1267.</p>
<p>Miller, J. M., Malenfant, R. M., David, P., Davis, C. S., Poissant,
J., Hogg, J. T., … &amp; Coltman, D. (2014). Estimating genome-wide
heterozygosity: effects of demographic history and marker type.
Heredity, 112(3), 240-247.</p>
<p>Nei, M. (1987). Molecular evolutionary genetics. Columbia university
press.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Keaka Farleigh, Mason Murphy, Christopher Blair, Tereza Jezkova.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>PopGenHelpR is funded by an National Science Foundation (NSF) Graduate Research Fellowship to KF (Award 2037786), an NSF grant to CB (DEB-1929679), and an NSF career grant to TJ (DEB-2144047).</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
